internal/build/job/maker.go

```diff
@@ -18,13 +14,62 @@ import (
 	"k8s.io/utils/pointer"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/build"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/constants"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
 )
 
+const (
+	dockerfileVolumeName = "dockerfile"
+)
+
+type jobTemplateOptions struct {
+	containerImage string
+
+	registryTLS *kmmv1beta1.TLSOptions
+
+	owner metav1.Object
+}
+
+// JobTemplateOption configures MakeJobTemplate.
+type JobTemplateOption func(*jobTemplateOptions)
+
+// ContainerImage configures the image to be pushed. When left
+// empty no image is pushed.
+func ContainerImage(containerImage string) JobTemplateOption {
+	return func(opts *jobTemplateOptions) {
```

Comment:
Why are we returning a function instead of the object? Why are we delaying the execution of `opts.containerImage = containerImage` to when the function will be executed rather than just now?